src/main/scala/vidispine/VSCommunicator.scala

```diff
@@ -0,0 +1,69 @@
+package vidispine
+
+import java.util.Base64
+
+import actors.messages.ErrorSend
+import akka.event.DiagnosticLoggingAdapter
+import akka.stream.scaladsl.{Keep, Sink, Source}
+import akka.util.ByteString
+import com.softwaremill.sttp._
+
+import scala.concurrent.{ExecutionContext, Future}
+import scala.util.{Failure, Success, Try}
+
+trait VSCommunicator {
+  protected val plutoUser:String
+  protected val plutoPass:String
+
+  private def authString:String = Base64.getEncoder.encodeToString(s"$plutoUser:$plutoPass".getBytes)
+
+  protected implicit val sttpBackend:SttpBackend[Future,Source[ByteString, Any]]
+  protected implicit val logger:DiagnosticLoggingAdapter
+
+  private def sendPut(uri:Uri, xmlString:String, headers:Map[String,String]):Future[Response[Source[ByteString, Any]]] = {
+    val bs = ByteString(xmlString,"UTF-8")
+
+    logger.info("Got xml body")
+    val source:Source[ByteString, Any] = Source.single(bs)
+    val hdr = Map(
+      "Accept"->"application/xml",
+      "Authorization"->s"Basic $authString",
+      "Content-Type"->"application/xml"
+    ) ++ headers
+
+    logger.info(hdr.toString())
+    logger.info(s"Got headers, initiating send to $uri")
+    sttp
+      .put(uri)
+      .streamBody(source)
+      .headers(hdr)
+      .response(asStream[Source[ByteString, Any]])
+      .send()
+  }
+
+  private def consumeSource(source:Source[ByteString,Any])(implicit logger:DiagnosticLoggingAdapter, materializer: akka.stream.Materializer, ec: ExecutionContext):Future[String] = {
```

Comment:
I haven't use the wrapper library that you're using here, but with plain akka-http this is quite simple - https://doc.akka.io/docs/akka-http/current/scala/http/implications-of-streaming-http-entity.html#consuming-the-http-response-entity-client-
e.g. `response.entity.toStrict(3.seconds)`